### DISCUSSION TITLE
How can i use this to Django serializer (Rest Framework)

### DISCUSSION THREAD

**Discussion by shahensaifullah** [fabiocaccamo/django-treenode#79] [Q&A]
I want to use this in django rest framework with serializer. How can i use this?

**Comment by fabiocaccamo** [comment]
@shahriar350 I'm sorry, I don't use DRF, consequently I've never tested this library with it. Maybe someone else did it and can give you a more useful answer.

**Reply by shahensaifullah** [reply]
I mainly work with DRF. So, i need same type of package for DRF which can make tree based recursive relational data.

**Reply by baseplate-admin** [reply]
For which i think a 3rd party package will be better.. ( which doesn't exist at this point )

**Reply by luyaogod** [reply]
I use pydantic instead drf serializer for nested struct.
```python
from pydantic import BaseModel

class CategorySerializer(BaseModel):
    name: str
    children: list['CategorySerializer'] = []
```

**Reply by fabiocaccamo** [reply]
I use `django-ninja` and it's very easy to achieve this.
